scripts/submit-analytics.js

```diff
@@ -122,6 +121,40 @@ function getRealmVersion() {
   return packageJson["version"];
 }
 
+/**
+ * Reads and parses `dependencies.list`.
+ * Each line has be form "KEY=VALUE", and we to find "REALM_CORE_VERSION"
+ *
+ * @returns the Realm Core version as a string
+ */
+function getRealmCoreVersion() {
+  const dependenciesListPath = path.resolve(__dirname, "../dependencies.list");
+  const dependenciesList = fse
+    .readFileSync(dependenciesListPath)
+    .toString()
+    .split("\n")
+    .map((s) => s.split("="));
+  return dependenciesList.find((e) => e[0] === "REALM_CORE_VERSION")[0];
+}
+
+/**
+ * Determines if `npm` or `yarn` is used.
+ *
+ * @returns An array with two elements: method and version
+ */
+function getInstallationMethod() {
+  const root = getProjectRoot();
+  if (fse.exists(path.resolve(root, "yarn.lock"))) {
+    const yarnVersion = execSync("yarn --version").toString().trim();
+    return ["yarn", yarnVersion];
+  }
+  if (fse.exists(path.resolve(root, "package-lock.json"))) {
+    const npmVersion = execSync("npm --version").toString().trim();
+    return ["npm", npmVersion];
+  }
+  return ["unknown", "unknown"];
```

Comment:
I don't think this is reliable:
1. The repository might have a `yarn.lock` and and the user might have `yarn` installed, but have used npm to install
2. Even worse, the repo might have a `yarn.lock` but the user does not have `yarn` installed, in which case `execSync` will throw and the entire submit will fail.

Instead, I suggest using the `npm_config_user_agent` environment variable to determine this:
- Using `npm`: `npm/8.19.1 node/v18.9.0 darwin arm64 workspaces/false`
- Using `yarn`: `yarn/1.22.19 npm/? node/v18.9.0 darwin arm64`

Reply:
Cool, I will parse `npm_config_user_agent` instead of adding an extra dependency.

Reply:
A regular expression should do the trick 👍